test/java/seedu/addressbook/commands/AddCommandTest.java

```diff
@@ -0,0 +1,192 @@
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.person.UniquePersonList.DuplicatePersonException;
+import seedu.addressbook.data.tag.UniqueTagList;
+
+public class AddCommandTest {
+	private static final List<? extends ReadOnlyPerson> LAST_SHOWN_LIST = Collections.emptyList();
+	
+	/*
+	 * Note how the names of the test methods does not follow the normal naming
+	 * convention. That is because our coding standard allows a different naming
+	 * convention for test methods.
+	 */
+
+	@Test
+	public void addCommand_invalidName_throws() {
+		Set<String> tags = new HashSet<String>();
+		final String[] invalidNames = { "", " ", "'", "[]\\[;]" };
+		for (String name : invalidNames) {
+			ConstructingInvalidAddCmdThrows(name, Phone.EXAMPLE, false, Email.EXAMPLE, false, Address.EXAMPLE, false,
+					tags);
+		}
+	}
+
+	@Test
+	public void addCommand_invalidPhone_throws() {
+		Set<String> tags = new HashSet<String>();
+		final String[] invalidNumbers = { "", " ", "1234-5678", "[]\\[;]", "abc", "a123", "+651234" };
+		for (String number : invalidNumbers) {
+			ConstructingInvalidAddCmdThrows(Name.EXAMPLE, number, false, Email.EXAMPLE, false, Address.EXAMPLE, false,
+					tags);
+		}
+	}
+
+	@Test
+	public void addCommand_invalidEmail_throws() {
+		Set<String> tags = new HashSet<String>();
+		final String[] invalidEmails = { "", " ", "def.com", "@", "@def", "@def.com", "abc@" };
+		for (String email : invalidEmails) {
+			ConstructingInvalidAddCmdThrows(Name.EXAMPLE, Phone.EXAMPLE, false, email, false, Address.EXAMPLE, false,
+					tags);
+		}
+	}
+
+	@Test
+	public void addCommand_invalidAddress_throws() {
+		Set<String> tags = new HashSet<String>();
+		final String[] invalidAddresses = { "", " " };
+		for (String address : invalidAddresses) {
+			ConstructingInvalidAddCmdThrows(Name.EXAMPLE, Phone.EXAMPLE, false, Email.EXAMPLE, false, address, false,
+					tags);
+		}
+	}
+
+	@Test
+	public void addCommand_invalidTags_throws() {
+		final String[][] invalidTags = { { "" }, { " " }, { "'" }, { "validTag", "" }, { "", " " } };
+		for (String[] tags : invalidTags) {
+			Set<String> tagsToAdd = new HashSet<String>();
+			for (String tag : tags) {
+				tagsToAdd.add(tag);
+			}
+			ConstructingInvalidAddCmdThrows(Name.EXAMPLE, Phone.EXAMPLE, false, Email.EXAMPLE, false, Address.EXAMPLE,
+					false, tagsToAdd);
+		}
+	}
+
+	/**
+	 * Asserts that attempting to construct an add command with the supplied
+	 * invalid data throws an IllegalValueException
+	 */
+	private void ConstructingInvalidAddCmdThrows(String name, String phone, boolean isPhonePrivate, String email,
```

Comment:
We usually name this kind of methods with `assert...` to indicate that the method has an assert-like behavior inside it. e.g `assertConstructingInvalidAddCommandThrowsException`